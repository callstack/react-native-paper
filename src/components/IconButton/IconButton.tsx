import * as React from 'react';
import {
  GestureResponderEvent,
  StyleProp,
  StyleSheet,
  ViewStyle,
  View,
  Animated,
  ColorValue,
} from 'react-native';

import { getIconButtonColor } from './utils';
import { useInternalTheme } from '../../core/theming';
import type { $RemoveChildren, ThemeProp } from '../../types';
import { forwardRef } from '../../utils/forwardRef';
import ActivityIndicator from '../ActivityIndicator';
import CrossFadeIcon from '../CrossFadeIcon';
import Icon, { IconSource } from '../Icon';
import Surface from '../Surface';
import TouchableRipple from '../TouchableRipple/TouchableRipple';

const PADDING = 8;

type IconButtonMode = 'outlined' | 'contained' | 'contained-tonal';

export type Props = Omit<$RemoveChildren<typeof TouchableRipple>, 'style'> & {
  /**
   * Icon to display.
   */
  icon: IconSource;
  /**
   * @supported Available in v5.x with theme version 3
   * Mode of the icon button. By default there is no specified mode - only pressable icon will be rendered.
   */
  mode?: IconButtonMode;
  /**
   * @renamed Renamed from 'color' to 'iconColor' in v5.x
   * Color of the icon.
   */
  iconColor?: string;
  /**
   * Background color of the icon container.
   */
  containerColor?: string;
  /**
   * Color of the ripple effect.
   */
  rippleColor?: ColorValue;
  /**
   * @supported Available in v5.x with theme version 3
   * Whether icon button is selected. A selected button receives alternative combination of icon and container colors.
   */
  selected?: boolean;
  /**
   * Size of the icon.
   */
  size?: number;
  /**
   * Whether the button is disabled. A disabled button is greyed out and `onPress` is not called on touch.
   */
  disabled?: boolean;
  /**
   * Whether an icon change is animated.
   */
  animated?: boolean;
  /**
   * Accessibility label for the button. This is read by the screen reader when the user taps the button.
   */
  accessibilityLabel?: string;
  /**
   * Style of button's inner content.
   * Use this prop to apply custom height and width or to set a custom padding`.
   */
  contentStyle?: StyleProp<ViewStyle>;
  /**
   * Function to execute on press.
   */
  onPress?: (e: GestureResponderEvent) => void;
  style?: Animated.WithAnimatedValue<StyleProp<ViewStyle>>;
  ref?: React.RefObject<View>;
  /**
   * TestID used for testing purposes
   */
  testID?: string;
  /**
   * @optional
   */
  theme?: ThemeProp;
  /**
   * Whether to show a loading indicator.
   */
  loading?: boolean;
};

/**
 * An icon button is a button which displays only an icon without a label.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { IconButton, MD3Colors } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <IconButton
 *     icon="camera"
 *     iconColor={MD3Colors.error50}
 *     size={20}
 *     onPress={() => console.log('Pressed')}
 *   />
 * );
 *
 * export default MyComponent;
 * ```
 *
 * @extends TouchableRipple props https://callstack.github.io/react-native-paper/docs/components/TouchableRipple
 */
const IconButton = forwardRef<View, Props>(
  (
    {
      icon,
      iconColor: customIconColor,
      containerColor: customContainerColor,
      rippleColor: customRippleColor,
      size = 24,
      accessibilityLabel,
      disabled,
      onPress,
      selected = false,
      animated = false,
      mode,
      style,
      theme: themeOverrides,
      testID = 'icon-button',
      loading = false,
      contentStyle,
      ...rest
    }: Props,
    ref
  ) => {
    const theme = useInternalTheme(themeOverrides);
    const { isV3 } = theme;

    const IconComponent = animated ? CrossFadeIcon : Icon;

    const { iconColor, rippleColor, backgroundColor, borderColor } =
      getIconButtonColor({
        theme,
        disabled,
        selected,
        mode,
        customIconColor,
        customContainerColor,
        customRippleColor,
      });

    const buttonSize = isV3 ? size + 2 * PADDING : size * 1.5;

    const {
      borderWidth = isV3 && mode === 'outlined' && !selected ? 1 : 0,
      borderRadius = buttonSize / 2,
    } = (StyleSheet.flatten(style) || {}) as ViewStyle;

    const borderStyles = {
      borderWidth,
      borderRadius,
      borderColor,
    };

    return (
      <Surface
        ref={ref}
        testID={`${testID}-container`}
        style={[
          {
            backgroundColor,
            width: buttonSize,
            height: buttonSize,
          },
          styles.container,
          borderStyles,
          !isV3 && disabled && styles.disabled,
          style,
        ]}
        container
        {...(isV3 && { elevation: 0 })}
      >
        <TouchableRipple
          borderless
          centered
          onPress={onPress}
          rippleColor={rippleColor}
          accessibilityLabel={accessibilityLabel}
          style={[styles.touchable, contentStyle]}
          // @ts-expect-error We keep old a11y props for backwards compat with old RN versions
          accessibilityTraits={disabled ? ['button', 'disabled'] : 'button'}
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={{ disabled }}
          disabled={disabled}
          hitSlop={
            TouchableRipple.supported
              ? { top: 10, left: 10, bottom: 10, right: 10 }
              : { top: 6, left: 6, bottom: 6, right: 6 }
          }
          testID={testID}
          {...rest}
        >
          {loading ? (
            <ActivityIndicator size={size} color={iconColor} />
          ) : (
            <IconComponent color={iconColor} source={icon} size={size} />
          )}
        </TouchableRipple>
      </Surface>
    );
  }
);

const styles = StyleSheet.create({
  container: {
    overflow: 'hidden',
    margin: 6,
    elevation: 0,
  },
  touchable: {
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  disabled: {
    opacity: 0.32,
  },
});

export default IconButton;
