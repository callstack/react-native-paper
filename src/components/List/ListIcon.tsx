import * as React from 'react';
import { View, ViewStyle, StyleSheet, StyleProp } from 'react-native';

import { useInternalTheme } from '../../core/theming';
import type { ThemeProp } from '../../types';
import Icon, { IconSource } from '../Icon';

export type Props = {
  /**
   * Icon to show.
   */
  icon: IconSource;
  /**
   * Color for the icon.
   */
  color?: string;
  style?: StyleProp<ViewStyle>;
  /**
   * @optional
   */
  theme?: ThemeProp;
  /**
   * Size of the icon.
   */
  size?: number;
};

const ICON_SIZE = 24;

/**
 * A component to show an icon in a list item.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { List, MD3Colors } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <>
 *     <List.Icon color={MD3Colors.tertiary70} icon="folder" />
 *     <List.Icon color={MD3Colors.tertiary70} size={20} icon="equal" />
 *     <List.Icon color={MD3Colors.tertiary70} icon="calendar" />
 *   </>
 * );
 *
 * export default MyComponent;
 * ```
 */
const ListIcon = ({
  icon,
  color: iconColor,
  style,
  theme: themeOverrides,
  size,
}: Props) => {
  const theme = useInternalTheme(themeOverrides);
  const iconSize = size ? size : ICON_SIZE;

  return (
    <View
      style={[theme.isV3 ? styles.itemV3 : styles.item, style]}
      pointerEvents="box-none"
    >
      <Icon source={icon} size={iconSize} color={iconColor} theme={theme} />
    </View>
  );
};

const styles = StyleSheet.create({
  item: {
    margin: 8,
    height: 40,
    width: 40,
    alignItems: 'center',
    justifyContent: 'center',
  },
  itemV3: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});

ListIcon.displayName = 'List.Icon';

export default ListIcon;
