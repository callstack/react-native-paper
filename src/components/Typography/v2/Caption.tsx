import * as React from 'react';
import { Text, TextStyle, StyleSheet, StyleProp } from 'react-native';

import StyledText from './StyledText';

export type Props = React.ComponentProps<typeof Text> & {
  style?: StyleProp<TextStyle>;
  children: React.ReactNode;
};

// @component-group Typography
/**
 * @deprecated Deprecated in v5.x - use `<Text variant="bodySmall" />` instead.
 * Typography component for showing a caption.
 *
 * <div class="screenshots">
 *   <img src="screenshots/caption.png" />
 * </div>
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { Caption } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <Caption>Caption</Caption>
 * );
 *
 * export default MyComponent;
 * ```
 */
const Caption = (props: Props) => (
  <StyledText
    {...props}
    alpha={0.54}
    family="regular"
    style={[styles.text, props.style]}
  />
);

export default Caption;

const styles = StyleSheet.create({
  text: {
    fontSize: 12,
    lineHeight: 20,
    marginVertical: 2,
    letterSpacing: 0.4,
  },
});
