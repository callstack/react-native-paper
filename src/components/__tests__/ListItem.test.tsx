import * as React from 'react';
import { Platform, StyleSheet } from 'react-native';
import { Text, View } from 'react-native';

import { fireEvent, render } from '@testing-library/react-native';

import { red500 } from '../../styles/themes/v2/colors';
import Chip from '../Chip/Chip';
import IconButton from '../IconButton/IconButton';
import ListIcon from '../List/ListIcon';
import ListItem from '../List/ListItem';

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
  },
  description: {
    color: red500,
  },
  content: {
    paddingLeft: 0,
  },
});

const testID = 'list-item';

it('renders list item with title and description', () => {
  const tree = render(
    <ListItem
      title="First Item"
      testID={testID}
      description="Description for first item"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with left item', () => {
  const tree = render(
    <ListItem
      title="First Item"
      testID={testID}
      left={(props) => <ListIcon {...props} icon="folder" />}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with right item', () => {
  const tree = render(
    <ListItem
      title="First Item"
      testID={testID}
      right={() => <Text>GG</Text>}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with left and right items', () => {
  const tree = render(
    <ListItem
      title="First Item"
      description="Item description"
      testID={testID}
      left={() => <Text>GG</Text>}
      right={(props) => <ListIcon {...props} icon="folder" />}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with custom title and description styles', () => {
  const tree = render(
    <ListItem
      title="First Item"
      description="Item description"
      testID={testID}
      titleStyle={styles.title}
      descriptionStyle={styles.description}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with custom description', () => {
  const tree = render(
    <ListItem
      title="List Item with custom description"
      description={({ ellipsizeMode, color: descriptionColor, fontSize }) => (
        <View>
          <Text
            numberOfLines={2}
            ellipsizeMode={ellipsizeMode}
            style={{ color: descriptionColor, fontSize }}
          >
            React Native Paper is a high-quality, standard-compliant Design
            Design library that has you covered in all major use-cases.
          </Text>
          <View>
            <Chip icon="file-pdf-box" onPress={() => {}}>
              DOCS.pdf
            </Chip>
          </View>
        </View>
      )}
      testID={testID}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders with a description with typeof number', () => {
  const tree = render(
    <ListItem
      title="First Item"
      description={123}
      titleStyle={styles.title}
      descriptionStyle={styles.description}
      testID={testID}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('calling onPress on ListItem right component', () => {
  Platform.OS = 'web';
  const onPress = jest.fn();

  const { getByTestId } = render(
    <ListItem
      title="First Item"
      description="Item description"
      testID={testID}
      right={() => <IconButton icon="pencil" onPress={onPress} />}
    />
  );

  fireEvent(getByTestId('icon-button'), 'onPress');
  expect(onPress).toHaveBeenCalledTimes(1);
});

it('renders list item with custom content style', () => {
  const { getByTestId } = render(
    <ListItem
      title="First Item"
      description="Item description"
      contentStyle={styles.content}
      testID={testID}
    />
  );

  expect(getByTestId('list-item-content')).toHaveStyle(styles.content);
});
