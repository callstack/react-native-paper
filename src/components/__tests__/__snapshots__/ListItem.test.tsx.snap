// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list item with custom description 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        List Item with custom description
      </Text>
      <View>
        <Text
          numberOfLines={2}
          style={
            {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 14,
            }
          }
        >
          React Native Paper is a high-quality, standard-compliant Design Design library that has you covered in all major use-cases.
        </Text>
        <View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(232, 222, 248, 1)",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(232, 222, 248, 1)",
                  "borderColor": "transparent",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "alignSelf": "center",
                          "padding": 4,
                        },
                        {
                          "paddingLeft": 8,
                          "paddingRight": 0,
                        },
                        null,
                      ]
                    }
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontSize": 18,
                          },
                          [
                            {
                              "lineHeight": 18,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                        ]
                      }
                    >
                      file-pdf-box
                    </Text>
                  </View>
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(29, 25, 43, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 8,
                              "marginRight": 16,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    DOCS.pdf
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list item with custom title and description styles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              {
                "fontSize": 20,
              },
            ],
          ]
        }
      >
        First Item
      </Text>
      <Text
        numberOfLines={2}
        onTextLayout={[Function]}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 14,
              },
              {
                "color": "rgba(73, 69, 79, 1)",
              },
              {
                "color": "#f44336",
              },
            ],
          ]
        }
      >
        Item description
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders list item with left and right items 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text>
      GG
    </Text>
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        First Item
      </Text>
      <Text
        numberOfLines={2}
        onTextLayout={[Function]}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 14,
              },
              {
                "color": "rgba(73, 69, 79, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        Item description
      </Text>
    </View>
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "alignSelf": "center",
            "marginLeft": 16,
            "marginRight": 0,
          },
        ]
      }
    >
      <Text
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
          ]
        }
      >
        folder
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders list item with left item 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "alignSelf": "center",
            "marginLeft": 16,
            "marginRight": 0,
            "marginVertical": 0,
          },
        ]
      }
    >
      <Text
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
          ]
        }
      >
        folder
      </Text>
    </View>
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        First Item
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders list item with right item 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        First Item
      </Text>
    </View>
    <Text>
      GG
    </Text>
  </View>
</View>
`;

exports[`renders list item with title and description 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        First Item
      </Text>
      <Text
        numberOfLines={2}
        onTextLayout={[Function]}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 14,
              },
              {
                "color": "rgba(73, 69, 79, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        Description for first item
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders with a description with typeof number 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        undefined,
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "marginVertical": 6,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              {
                "fontSize": 20,
              },
            ],
          ]
        }
      >
        First Item
      </Text>
      <Text
        numberOfLines={2}
        onTextLayout={[Function]}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 14,
              },
              {
                "color": "rgba(73, 69, 79, 1)",
              },
              {
                "color": "#f44336",
              },
            ],
          ]
        }
      >
        123
      </Text>
    </View>
  </View>
</View>
`;
